accounts/backend.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2017 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package accounts
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+// Backend is an "account provider" that can specify a batch of accounts it can
+// sign transactions with and upon request, do so.
+type Backend interface {
+	// Accounts retrieves the list of signing accounts the backend is currently aware of.
```

Comment:
The same way we do it for keystores. The backend maintains a local cache of accounts it's currently aware of and returns that. The keystore registeres filesystem watches and updates its cache on change (or periodically on android). The same can be done for HW wallets, where the backend monitors USB events, and maintains a list of supported devices.